File: platform/polyglot/fmt_sdk/go/logger.go
package fmt_sdk

import (
	"log/slog"
	"os"
	"path/filepath"
)

// Logger wraps slog.Logger with pre-configured fields for package and file.
type Logger struct {
	inner *slog.Logger
}

// CreateLogger creates a Logger that writes JSON to stderr with pkg and file attributes.
func CreateLogger(pkg, file string) *Logger {
	handler := slog.NewJSONHandler(os.Stderr, &slog.HandlerOptions{
		Level: slog.LevelDebug,
	})
	return &Logger{
		inner: slog.New(handler).With("pkg", pkg, "file", filepath.Base(file)),
	}
}

// CreateLoggerWith creates a Logger using the provided slog.Handler.
func CreateLoggerWith(pkg, file string, handler slog.Handler) *Logger {
	return &Logger{
		inner: slog.New(handler).With("pkg", pkg, "file", filepath.Base(file)),
	}
}

// Info logs at INFO level.
func (l *Logger) Info(msg string, args ...any) { l.inner.Info(msg, args...) }

// Warn logs at WARN level.
func (l *Logger) Warn(msg string, args ...any) { l.inner.Warn(msg, args...) }

// Error logs at ERROR level.
func (l *Logger) Error(msg string, args ...any) { l.inner.Error(msg, args...) }

// Debug logs at DEBUG level.
func (l *Logger) Debug(msg string, args ...any) { l.inner.Debug(msg, args...) }

File: platform/polyglot/fmt_sdk/rs/src/logger.rs
use log::{debug, error, info, warn};
use std::path::Path;

pub struct Logger {
    pkg: String,
    file: String,
}

impl Logger {
    pub fn create(pkg: &str, file: &str) -> Self {
        let filename = Path::new(file)
            .file_name()
            .and_then(|f| f.to_str())
            .unwrap_or("unknown")
            .to_string();
        Logger {
            pkg: pkg.to_string(),
            file: filename,
        }
    }

    pub fn info(&self, msg: &str) {
        info!("[{}:{}] {}", self.pkg, self.file, msg);
    }

    pub fn warn(&self, msg: &str) {
        warn!("[{}:{}] {}", self.pkg, self.file, msg);
    }

    pub fn error(&self, msg: &str) {
        error!("[{}:{}] {}", self.pkg, self.file, msg);
    }

    pub fn debug(&self, msg: &str) {
        debug!("[{}:{}] {}", self.pkg, self.file, msg);
    }
}
